src/librustdoc/clean/types.rs

```diff
@@ -641,6 +643,53 @@ impl Item {
             _ => false,
         }
     }
+
+    crate fn hir_id(&self) -> hir::HirId {
+        hir::HirId::make_owner(
+            self.def_id.as_def_id().unwrap().as_local().expect("not local DefId"),
+        )
+    }
+
+    crate fn header(&self, tcx: TyCtxt<'_>) -> Option<hir::FnHeader> {
```

Comment:
Would be more clear to name this `fn_header`, and to document that it panics for non-local items.

Reply:
Good idea!